src/Metadata/RouteBasedResourceMetadataFactory.php

```diff
@@ -72,7 +76,8 @@ public function createMetadata(string $requestedName, array $metadata) : Abstrac
             $metadata['extractor'],
             $metadata['resource_identifier'] ?? 'id',
             $metadata['route_identifier_placeholder'] ?? 'id',
-            $metadata['route_params'] ?? []
+            $metadata['route_params'] ?? [],
+            $metadata['max_depth'] ?? 10
```

Comment:
Since we don't know exactly where the config is coming from, it may be a good idea to do something like:

```php
$maxDepth = $metadata['max_depth'] ?? 10;
```

somewhere before this, and then, here:

```php
(int) $maxDepth
```

to ensure that there are not type issues.

(Many config formats will return everything as strings by default.)

Reply:
not sure if silently supporting a wrong type is the right way to go here.
i'd rather see an TypeError to inform me that i use a wrong type instead of continuing using the wrong but silently supported type.
IMHO this is just hiding a some kind of complexity (using string in config which is then "magically" (from an outer perspective) casted to int).

Reply:
ping @weierophinney 
any thoughts on this?

Reply:
Leave it as-is. Most of the config libraries we support will cast values as they are processed, before they are pushed into the container. And the primary config type is pure PHP.